data_structures/src/chain.rs

```diff
@@ -2088,6 +2088,14 @@ impl DataRequestState {
         };
         self.info.current_stage = Some(self.stage);
     }
+
+    /// Function to calculate the backup witnesses required
+    pub fn backup_witnesses(&self) -> u16 {
```

Comment:
This function should be better documented, maybe move it somewhere else as a function of `f(witnesses, commit_round) -> backup_witnesses` so it can be better tested.